pkg/providers/cloudstack/validator.go

```diff
@@ -26,27 +26,56 @@ func NewValidator(cmk ProviderCmkClient) *Validator {
 
 type ProviderCmkClient interface {
 	ValidateCloudStackConnection(ctx context.Context) error
-	ValidateServiceOfferingPresent(ctx context.Context, domain string, zone anywherev1.CloudStackResourceRef, account string, serviceOffering anywherev1.CloudStackResourceRef) error
-	ValidateTemplatePresent(ctx context.Context, domain string, zone anywherev1.CloudStackResourceRef, account string, template anywherev1.CloudStackResourceRef) error
-	ValidateAffinityGroupsPresent(ctx context.Context, domain string, zone anywherev1.CloudStackResourceRef, account string, affinityGroupIds []string) error
-	ValidateZonePresent(ctx context.Context, zone anywherev1.CloudStackResourceRef) error
+	ValidateServiceOfferingPresent(ctx context.Context, zoneId string, serviceOffering anywherev1.CloudStackResourceRef) error
+	ValidateTemplatePresent(ctx context.Context, domainId string, zoneId string, account string, template anywherev1.CloudStackResourceRef) error
+	ValidateAffinityGroupsPresent(ctx context.Context, domainId string, account string, affinityGroupIds []string) error
+	ValidateZonesPresent(ctx context.Context, zones []anywherev1.CloudStackZoneRef) ([]anywherev1.CloudStackResourceIdentifier, error)
+	ValidateNetworkPresent(ctx context.Context, domainId string, zoneRef anywherev1.CloudStackZoneRef, zones []anywherev1.CloudStackResourceIdentifier, account string, multipleZone bool) error
+	ValidateDomainPresent(ctx context.Context, domain string) (anywherev1.CloudStackResourceIdentifier, error)
+	ValidateAccountPresent(ctx context.Context, account string, domainId string) error
 }
 
 func (v *Validator) validateCloudStackAccess(ctx context.Context) error {
 	if err := v.cmk.ValidateCloudStackConnection(ctx); err != nil {
-		return fmt.Errorf("failed validating connection to vCenter: %v", err)
+		return fmt.Errorf("failed validating connection to cloudStack: %v", err)
 	}
 	logger.MarkPass("Connected to server")
 
 	return nil
 }
 
 func (v *Validator) ValidateCloudStackDatacenterConfig(ctx context.Context, datacenterConfig *anywherev1.CloudStackDatacenterConfig) error {
-	if err := v.cmk.ValidateZonePresent(ctx, datacenterConfig.Spec.Zone); err != nil {
-		return err
+	if len(datacenterConfig.Spec.Domain) <= 0 {
+		return fmt.Errorf("CloudStackDatacenterConfig domain is not set or is empty")
+	}
+	domain, errDomain := v.cmk.ValidateDomainPresent(ctx, datacenterConfig.Spec.Domain)
+	if errDomain != nil {
+		return fmt.Errorf("error while checking domain %v", errDomain)
+	}
+
+	if len(datacenterConfig.Spec.Account) > 0 {
+		err := v.cmk.ValidateAccountPresent(ctx, datacenterConfig.Spec.Account, domain.Id)
+		if err != nil {
+			return fmt.Errorf("error while checking account %v", err)
```

Comment:
No need to prepend 'error' to errors. Thats done a lot in these checks, can we strip that out.

```
invalid account: %v
```